src/apiflask/scaffold.py

```diff
@@ -515,7 +515,9 @@ def doc(
         summary: t.Optional[str] = None,
         description: t.Optional[str] = None,
         tags: t.Optional[t.List[str]] = None,
-        responses: t.Optional[t.Union[t.List[int], t.Dict[int, str]]] = None,
+        responses: t.Optional[
+            t.Union[t.List[int], t.Dict[int, str], t.Dict[int, t.Dict[str, t.Union[str, t.Dict]]]]
```

Comment:
When specifying a custom response, the schema can be a dict, schema class, or schema class instance. I would recommend moving this type definition to the `types.py` module.

Reply:
I'm a little confused by this feedback. From looking at the test cases in `test_doc_responses_custom_spec`, I see support for the schema and schema class instance, but those are nested within a couple layers of dictionaries. I left the final bit of typing information here as just `t.Dict`, since the prior structures are required by the OpenAPI spec (to include the `content` and media type elements), but there might be other things in this structure as well (`headers` or `links`, for instance). Are you suggesting I expand this out to something like the following:
```
t.Union[
    t.List[int],
    t.Dict[int, str],
    t.Dict[
        int,
        t.Dict[
            str, t.Union[str, t.Dict[str, t.Dict[str, t.Union[Schema, t.Type[Schema], t.Dict]]]]
        ],
    ],
]
```
... to account for the handling of the `schema` field? I think I'd need to take a closer look at the OpenAPI spec to ensure that matches all of the supported structures appropriately; I deliberately left it as an open-ended `Dict` because the spec is so complex.

Once we're agreed on the desired typing signature, I'll add it to the `types.py` module - I didn't realize it existed :).

I also realized that there's some typing info on the `responses` object in the `APIFlask._generate_spec` method; I'm going to update that with my current typing definition, and will update it with whatever we agree on as the right typing for the custom response object.

Reply:
Yes, `apispec` will "translate" the schema or schema instance in the nesting `responses` dict to the OpenAPI spec in the dict format.

Reply:
I took a look at the OpenAPI spec and have added typings for a Responses Object and a Response Object. Unfortunately, I wasn't able to incorporate schemas or schema classes into the typing definition because the OAS accepts a pretty wide range of data types. The initial definition after combing over the spec looked like this:

```
ReferenceObjectType = t.Dict[str, str]
MediaTypeObjectType = t.Dict[str, t.Any]
HeaderObjectType = t.Dict[str, t.Any]
LinkObjectType = t.Dict[str, t.Any]
HeadersObjectType = t.Dict[str, t.Union[HeaderObjectType, ReferenceObjectType]]
ContentObjectType = t.Dict[str, MediaTypeObjectType]
LinksObjectType = t.Dict[str, t.Union[LinkObjectType, ReferenceObjectType]]
ResponseObjectType = t.Dict[
    str, t.Union[str, HeadersObjectType, ContentObjectType, LinksObjectType]
]
ResponsesObjectType = t.Dict[t.Union[int, str], t.Union[ResponseObjectType, ReferenceObjectType]]
ResponsesType = t.Union[t.List[int], t.Dict[int, str], ResponsesObjectType]
```

... which I then scoped down to the simplified version included in my changes. Because the Media Type Object has an `example` field that is typed as `Any`, I didn't see much value in adding a schema or schema type to the typing definition, since those would be covered by `typing.Any` and the typing definition should accept anything deemed valid by the OAS.

Reply:
LGTM.